### DISCUSSION TITLE
Findings and thoughts about OpenRadioss

### DISCUSSION THREAD

**Discussion by lcheng9** [OpenRadioss/OpenRadioss#1124] [Users]
Hello, exploring the OpenRadioss solver and result modules for a few days, I have some thoughts would share here and hope OpenRadioss becomes more exceptional. Overall, OpenRadioss is a great open-source project, I believe that the entire simulation community can benefit from it. This project clearly has the potential to leverage simulation technology to drive more adaption in all sectors of human beings' R&D. 

Here are my findings and thoughts:
1. th_to_csv.exe executable works only on the /TFILE/4 generated T01 files. For /TFILE/1-3, the conversion fails. 
2. The converted vtk files using anim_to_vtk.exe contain duplicated node ids for Tet4 and Tri3 elements. It looks like the animation file or the conversion assumes the Hex/Quad element in the result files. Although this won't impact the result reading, resolving this can make the vtk files smaller.
3. Since Tet10 has been supported at some areas, SH6N or Tri6 element worth supporting as well. 
4. **/TH supports the contour results with min and max values at each step, such as the stress on a group of nodes**. This feature seems to have the highest priority for me due to its wide usability. I really would try this functionality in the near future.
5. expose an option to output the csv and pvtu files directly, instead of manual conversion afterward. I understand this feature could be difficult to be approved, if so please consider making the license of th_to_csv and anim_to_vtk more friendly, such as BSD, MIT, etc.

<s>6. /TH/TITLE command in the engine file fails the solve. </s>

Since I am still new to OpenRadioss, my thoughts above could be naive, and feel free to correct me. I will be keeping exploring OpenRadioss. Thank you.

**Comment by elequiniou** [comment]
Dear Icheng9,

Thanks a lot for your interest in the project and for the time taking to test it and report your interesting findings.
Regarding the technical points, I need to have them reviewed by some specialists in the dev team to come back to you with more detailed answer.
About point 5. While direct output could be supported natively in the future, I realize the current license for the conversion codes is a bit restrictive to bring them into external workflow. This is a fair point we will investigate. 

Best Regards,

Eric

**Comment by lcheng9** [comment]
@elequiniou Thank you so much for your kind response. Look forward to hearing from you in the future as well. Best regards.

**Comment by sebastienVilleneuve** [comment]
Hi @lcheng9 
 
Some answers to the different points 

1.	Most commonly used ( and default one) format for /TFILE , is 4 , but it can be a good way to enhance the converter to    
        support all formats ( even if I think other formats are less used than Binary IEEE 32-bit )

2.	Thanks for the remark , here for sure we can enhance this to optimize the file size .

4.	This request is also a really interesting one , thank you for this .
        We need to discuss this internally, It is something we may develop in the OpenRadioss code for the future.

5.     Relative to native output of csv & pvtu files & license , Eric already answered

6.	Can you give me more information , or a model to reproduce this ? ( can’t reproduce with models I can test) 

Thanks,

Regards,
Sébastien

**Comment by lcheng9** [comment]
Hello @sebastienVilleneuve, thank you so much for your answers and comments. Please allow me to respond here and elaborate the item 4.

Item 1. If the T01 converter works only for the /TFILE/4, I am totally Ok and can use the Binary IEEE format only. 

Item 2. Since I am using the A001 file directly, please consider optimizing the A001 file size instead. Based on the reading scheme in the anim_to_vtk, I guess the mesh data in the A001 is based on Hex/Quad even if the element is Tet/Tri. 

Item 3. The quadratic element has many advantages in the simulation, I understand that Tri6/SH6N may have some challenges in the contact and xfem, so not an instant feature, for me at least.

Item 4. Min/Max values are one of our major concerns in the simulation, when computation is done, we probably prefer to know the min/max of some key quantities along the history, such as displacement, stress, and strain. While loading all result files for such information could be very expensive, and sometimes unnecessary. Providing such lightweight history data can benefit end users and post-processing developers to query the preliminary results, particularly for explicit dynamics analyses. See the attached picture, the table and chart cannot display the min/max stress along the history for now without reading all A001 files. The useful min/max history data could be displacement/velocity/acceleration xyz and their magnitudes, nodal stresses, and nodal strains. The data could be about all nodes only, since a specific group of nodes may bring up performance concerns. Such min/max history data could be in another type of file if not suitable to be combined with T01. 

![radioss_welsim_post_processing](https://user-images.githubusercontent.com/106984354/208940198-f10e64ef-ae70-4258-9e93-7f5d4cd227cd.PNG)

Item 5. Yes, Eric and you have addressed it. 

Item 6. I accidentally placed the /TH/TITLE in the incorrect space as shown below. My bad, please skip this one.
```
/VERS/2022
/RUN/Radioss_WelSim/1/
/TH/TITLE
0.02
/RFILE
1000000
/ANIM/VECT/DISP
```

**Comment by elequiniou** [comment]
Update for point 5:
As announced at the OpenRadioss Users Day on June 27th, the converter tools are now under permissive license, they can be retrieved under https://github.com/OpenRadioss/Tools.

Notice this "Tools" repository is used for all the tools under permissive license, for example "user_lib SDK" to build users materials.

Thanks,

Eric
